### PR TITLE
[SP4] Fix instsys cert handling (bsc#1195220)

### PR DESCRIPTION
**Pull request by skazi0**
The new version of SUSEConnect doesn't support OpenSSL's "BEGIN TRUSTED
CERTIFICATE" format. Separate `extract` calls are needed to get certs
for "openssl" and "pem" directories.

## Problem

Failure to install custom SMT/RMT certificates during installation (with suseconnect-ng).

- [Bugzilla link](https://bugzilla.suse.com/show_bug.cgi?id=1195220)
- https://github.com/yast/yast-registration/pull/266

## Solution

Instead of extracting once and copying to "openssl" and "pem" directories, extract twice in respective formats.

## Testing

- *Tested manually*

### CODE REVIEW

File: src/lib/registration/ssl_certificate.rb [hunk 1]
```diff
@@ -81,33 +78,17 @@ def self.download(url, insecure: false)
...
-        files.each do |file|
-          # FileUtils.cp does not seem to allow copying the links without dereferencing them.
-          Yast::Execute.locally("cp", "--no-dereference", "--preserve=links", file, subdir)
-          memo << File.join(subdir, File.basename(file))
-        end
-      end
-
-      # Cleanup
-      FileUtils.rm_rf(TMP_CA_CERTS_DIR)
+        "--overwrite", File.join(CA_CERTS_DIR, "openssl"))
```
Comment (lslezak) [inline src/lib/registration/ssl_certificate.rb:84]: The goal of the code above is to keep the original content. The `/var/lib/ca-certificates` directory contains the initial trusted certificates. The `--overwrite` option causes removal of the previous content and losing the default SSL certificates.

That means we would have troubles accessing the other SSL secured sites.

Unfortunately after removing the `--overwrite` option the command fails with error `directory already exists`. So we need to extract the files into a temporary location and copy them to the correct target directory.

**Review by lslezak — COMMENTED** [review]
Unfortunately this deletes the original `/var/lib/ca-certificates` files...

File: src/lib/registration/ssl_certificate.rb (hunk 1 above)
Comment (skazi0) [reply src/lib/registration/ssl_certificate.rb:84]: OK. Then we need the copy code. I'll add it back.

Comment (skazi0) [reply src/lib/registration/ssl_certificate.rb]: done

**Review by lslezak — COMMENTED** [review]
Please bump the version and add a `.changes` entry, otherwise LGTM.

**Review by lslezak — APPROVED** [review]
LGTM, thanks for the fix!

(I have tested it, and it works fine!)

**Pull request merged**